test/Microsoft.AspNetCore.TestHost.Tests/ClientHandlerTests.cs

```diff
@@ -272,6 +272,63 @@ public async Task ExceptionAfterFirstWriteIsReported()
             Assert.IsType<InvalidOperationException>(ex.GetBaseException());
         }
 
+        [Fact]
+        public Task ExceptionFromOnStartingFirstWriteIsReported()
+        {
+            var handler = new ClientHandler(PathString.Empty, new DummyApplication(context =>
+            {
+                context.Response.OnStarting(() =>
+                {
+                    throw new InvalidOperationException(new string('a', 1024 * 32));
+                });
+                return context.Response.WriteAsync("Hello World");
+            }));
+            var httpClient = new HttpClient(handler);
+            return Assert.ThrowsAsync<InvalidOperationException>(() => httpClient.GetAsync("https://example.com/",
+                HttpCompletionOption.ResponseHeadersRead));
+        }
+
+        [Fact]
+        public Task ExceptionFromOnStartingWithNoWriteIsReported()
+        {
+            var handler = new ClientHandler(PathString.Empty, new DummyApplication(context =>
+            {
+                context.Response.OnStarting(() =>
+                {
+                    throw new InvalidOperationException(new string('a', 1024 * 32));
+                });
+                return Task.CompletedTask;
```

Comment:
Good amount of shared code between these methods. Consider making the at least the last 2 lines a shared methods between the 3 tests.

Reply:
IMO it’s better to have tests be isolated than DRY